lib/segment/src/common/flags/buffered_dynamic_flags.rs

```diff
@@ -19,15 +19,19 @@ pub(crate) struct BufferedDynamicFlags {
 
     /// Pending changes to the storage flags.
     buffer: Arc<RwLock<AHashMap<PointOffsetType, bool>>>,
+
+    /// Lock to prevent concurrent flush and drop
+    pub is_alive_flush_lock: Arc<Mutex<bool>>,
```

Comment:
After some consideration, I am not sure this will actually prevent trying to interact with an inexistent file/directory. In [`PayloadIndex::drop_index`](https://github.com/qdrant/qdrant/blob/80646ebba2bda62aad12461077cb33c11c910a4d/lib/segment/src/index/struct_payload_index.rs#L900) we actually remove the directory BEFORE dropping the index. 

This lock only makes the dropping wait for the end of the flush, but the files may already be deleted externally.

Please correct me if I missed something 🤔